### PR TITLE
feat: fincube #26 frontend integration with smart contract

### PR DESCRIPTION
**Pull request by Raad05**

### CODE REVIEW

File: frontend/src/pages/welcome.tsx [hunk 1]
```diff
@@ -1,8 +1,291 @@
...
+      await waitForTransactionReceipt(config, { hash });
+
+      alert("Proposal cancelled!");
+    } catch (e) {
+      console.log(e);
+    }
+  };
+
+  // setVotingDelay()
+  const setDelay = async (e: MouseEvent) => {
```
Comment (antonin686) [inline frontend/src/pages/welcome.tsx:154]: a state has already been declared with the same name at the top

File: frontend/src/pages/welcome.tsx [hunk 2]
```diff
@@ -1,8 +1,291 @@
...
+      await waitForTransactionReceipt(config, { hash });
+
+      alert("Delay set!");
+    } catch (e) {
+      console.log(e);
+    }
+  };
+
+  // setPeriod()
+  const setPeriod = async (e: MouseEvent) => {
```
Comment (antonin686) [inline frontend/src/pages/welcome.tsx:175]: a state has already been declared with the same name at the top

**Review by antonin686 — CHANGES_REQUESTED** [review]
Some issues have been commented, please check them.

File: frontend/src/pages/welcome.tsx (hunk 1 above)
Comment (Raad05) [reply frontend/src/pages/welcome.tsx]: Resolved this one

File: frontend/src/pages/welcome.tsx (hunk 2 above)
Comment (Raad05) [reply frontend/src/pages/welcome.tsx]: Resolved this one

**Comment by Raad05** [comment]
Added smart contract interaction features along with "success" and  "error" popups

**Review by antonin686 — APPROVED** [review]
LGTM

**Pull request merged**